src/components/ExpandedViewMode/ExpandedViewModeContent/ActivityHistory/ActivityHistory.js

```diff
@@ -16,100 +17,174 @@ class ActivityHistory extends Component {
       address: this.props.goalAddress,
     })
   }
+  componentDidUpdate() {
```

Comment:
This doesn't quite solve the issue. We need to retrigger `fetchGoalHistory` periodically (meaning at a time interval, such as 'every 3 seconds' but only while looking at this component.

Reply:
Use https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setInterval